services/base-service/models/base/model.js

```diff
@@ -395,31 +395,32 @@ class BaseModel extends FinEsDataModel {
    * @returns {Promise} resolves to elasticsearch result
    */
   async get(id, opts={}, index) {
-    let _source_excludes = true;
+    let _source_excludes = 'roles';
     if( opts.admin ) _source_excludes = false;
-    else if( opts.compact ) _source_excludes = 'compact';
 
     let identifier = id.replace(/^\//, '').split('/');
     identifier.shift();
     identifier = identifier.join('/');
     //console.log(`FinEsNestedModel.get(${identifier}) on ${this.readIndexAlias}`);
 
-    let result= await this.client.get(
+    let result = await this.client.exists({
+      index: this.readIndexAlias,
+      id: identifier
+    });
+
+    if( !result ) return null;
+
+    result = await this.client.get(
```

Comment:
@UcDust this call is basically replicated in `this.client_get`  I wonder if there is a better way for these two sperate gets to be formulated.

Reply:
@qjhart I refactored that code a bit in the last commit so that it uses that `client_get()` function. Added a param for `_source_excludes`.